python/cudf/cudf/core/dataframe.py

```diff
@@ -4002,9 +4004,7 @@ def scatter_by_map(self, map_index, map_size=None, keep_index=True):
                 "Use an integer array/column for better performance."
             )
 
-        if map_size is None:
-            map_size = map_index.unique_count()
-        else:
+        if kwargs.get("debug", False) is True and map_size is not None:
```

Comment:
Did flake8 complain about `== True` vs is True? My fear is if someone passes `debug=1` this will return False. May need to change this to `== 1`.

Reply:
Correct, flake8 was complaining about the same. Added a `==1` check instead.